manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
+
+  if $github_username != undef {
+    $sources[$github_username] = "https://github.com/${github_username}/${title}.git"
```

Comment:
You can always add the following to your ```~/.gitconfig```:
```
[url "git@github.com:"]
	pushInsteadOf = "https://github.com/"
```

Or if you want to be more specific you can add katello after it.